test/libgit2.jl

```diff
@@ -129,9 +130,18 @@ mktempdir() do dir
             try
                 @test isdir(path)
                 @test isfile(joinpath(path, LibGit2.Consts.HEAD_FILE))
+                @test LibGit2.isattached(repo)
             finally
                 finalize(repo)
             end
+
+            path = joinpath("garbagefakery", "Example.Bare")
+            try
+                LibGit2.GitRepo(path)
```

Comment:
add something like `error("unexpected")` here otherwise if it succeeds the tests will not trigger